# Drill - Describing Data
###  by Created Isaac Hembree

In [19]:
import pandas as pd
import numpy as np
import statistics

In [89]:
names=['Greg','Marcia','Peter','Jan','Bobby','Cindy','Oliver']
Brady = pd.DataFrame(index=names)
Brady['age']=[14, 12, 11, 10, 8, 6, 8]
Brady

age
Greg     14
Marcia   12
Peter    11
Jan      10
Bobby     8
Cindy     6
Oliver    8

### What are the mean, median, and mode of the kids' ages when they first appeared on the show? 

In [42]:
print("The mean age is {}.".format(np.mean(Brady['age']))) 
print("The median age is {}.".format(np.median(Brady['age']))) 
print("The mode age is {}.".format(statistics.mode(Brady['age']))) 

The mean age is 9.857142857142858.
The median age is 10.0.
The mode age is 8.


###  What are the variance, standard deviation, and standard error?

In [43]:
print("The variance is {}.".format(np.var(Brady['age']))) 
print("The standard deviation is {}.".format(np.std(Brady['age'],ddof=1 ))) 
print("The standsrd error is {}.".format(np.std(Brady['age'] ,ddof=1) / np.sqrt(len(Brady['age'])))) 

The variance is 6.408163265306122.
The standard deviation is 2.734262327610589.
The standsrd error is 1.0334540197243192.


### Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

For the central tendency, I would pick mean because it best describes where the ages cluster around. I would pick variance because it's so low.

### Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

In [66]:
Brady.loc['Cindy', 'age']=7
print("The mean age is {}.".format(np.mean(Brady['age']))) 
print("The median age is {}.".format(np.median(Brady['age']))) 
print("The mode age is {}.".format(statistics.mode(Brady['age']))) 
print("The variance is {}.".format(np.var(Brady['age']))) 
print("The standard deviation is {}.".format(np.std(Brady['age'],ddof=1 ))) 
print("The standsrd error is {}.".format(np.std(Brady['age'] ,ddof=1) / np.sqrt(len(Brady['age']))))

The mean age is 10.0.
The median age is 10.0.
The mode age is 8.
The variance is 5.428571428571429.
The standard deviation is 2.516611478423583.
The standsrd error is 0.9511897312113418.


The mean and median age are now the same. The mode stay the same, and the variance went down. 

### Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

In [94]:
names=['Greg','Marcia','Peter','Jan','Bobby','Cindy','Jessica']
Brady = pd.DataFrame(index=names)
Brady['age']=[14, 12, 11, 10, 8, 7, 1]

In [93]:
print("The mean age is {}.".format(np.mean(Brady['age']))) 
print("The median age is {}.".format(np.median(Brady['age'])))
print("There is no mode for age")
print("The variance is {}.".format(np.var(Brady['age']))) 
print("The standard deviation is {}.".format(np.std(Brady['age'],ddof=1 ))) 
print("The standsrd error is {}.".format(np.std(Brady['age'] ,ddof=1) / np.sqrt(len(Brady['age']))))

The mean age is 9.0.
The median age is 10.0.
There is no mode age
The variance is 15.428571428571429.
The standard deviation is 4.242640687119285.
The standsrd error is 1.6035674514745462.


This changing my choice of central tendency to median. 

### On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

### Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

In [96]:
print((20+23+17+5)/4)

16.25


Based on this numbers, I would use the mean to estimate the percentage of fan, which is 16.25%.